build.gradle

```diff
@@ -38,7 +38,7 @@ gen.mkdirs();
 
 task updateLibs(type: Download) {
 	src([
-		'https://yivesmirror.com/files/spigot/spigot-1.12.2-R0.1-SNAPSHOT-b1408.jar',
+		'https://yivesmirror.com/files/paperspigot/PaperSpigot-1.12.2-b1244.jar',
```

Comment:
You don't need it, when accessing fields with reflection.

Do you execute methods that needs it as a dependency?

Reply:
reflection is slow, although that would work. there's got to be a better way though.

Reply:
@ysl3000 not everything uses Reflection, that's why it needs to be compiled against Paper

Reply:
...to be honest I'm not even sure if it actually used any Paper stuff

Reply:
That wasn't the intention of this my comment.

At one point you're using reflection. The rest of your code seems not to access any variables/methods that requires Paperspigot.

Reply:
Yeah, I need to double check if it can be compiled against Spigot, I think I was compiling against Paper for testing purposes (and because I was decompiling the NMS classes via IntelliJ IDEA's class decompiler) and I never switched back to Spigot to test if it would actually compile against it.

If it does compile, I will fix that and the other stuff you mentioned 😉

Reply:
The intention was to show you that the code doesn't need paperSpigot to be executed.
So it can be merged, when changes are done.
Revert the dependency changes.

Reply:
@ysl3000 yeah I know, that's what I was saying.

I just never compiled against Spigot because I thought my code had some Paper specific stuff, but it seems that isn't the case after relooking the code.

Reply:
After fighting gradle because for some random reason it wasn't finding the RakNetServer dependency, it seems that it *does* use Paper specific stuff

https://i.imgur.com/N98Ho7S.png

However it can be replaced with Reflection

Reply:
Looking into it